compiler/rustc_lint/src/context/diagnostics.rs

```diff
@@ -427,15 +427,22 @@ pub(super) fn builtin(
                 db.note("see <https://doc.rust-lang.org/nightly/unstable-book/compiler-flags/check-cfg.html> for more information about checking conditional configuration");
             }
         }
-        BuiltinLintDiagnostics::DeprecatedWhereclauseLocation(new_span, suggestion) => {
-            db.multipart_suggestion(
-                "move it to the end of the type declaration",
-                vec![(db.span.primary_span().unwrap(), "".to_string()), (new_span, suggestion)],
-                Applicability::MachineApplicable,
-            );
-            db.note(
-                        "see issue #89122 <https://github.com/rust-lang/rust/issues/89122> for more information",
-                    );
+        BuiltinLintDiagnostics::DeprecatedWhereclauseLocation(sugg) => {
```

Comment:
It's sad that I couldn't consolidate the messages of the lint and the error but the bridging between `rustc_ast_passes` & `rustc_lint` is too tricky. If `rustc_lint_defs` were to have a `messages.ftl` I could make both crates import the `fluent_generated` messages but that's currently not the case.